Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,191 @@
+        /// </summary>
+        public static readonly DependencyProperty HeaderLineBrushProperty = DependencyProperty.Register(
+            nameof(HeaderLineBrush),
+            typeof(Brush),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(Application.Current.Resources["SystemControlBackgroundAccentBrush"]));
+
+        /// <summary>
+        /// Gets or sets the DeviceList Selection Mode. Defaults to ListViewSelectionMode.Single
+        /// </summary>
+        public ListViewSelectionMode SelectionMode
+        {
+            get { return (ListViewSelectionMode)GetValue(SelectionModeProperty); }
+            set { SetValue(SelectionModeProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="SelectionMode"/>.
+        /// </summary>
+        public static readonly DependencyProperty SelectionModeProperty = DependencyProperty.Register(
+            nameof(SelectionMode),
+            typeof(ListViewSelectionMode),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(ListViewSelectionMode.Single));
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RemoteDevicePicker"/> class.
+        /// </summary>
+        public RemoteDevicePicker()
+        {
+            DefaultStyleKey = typeof(RemoteDevicePicker);
+            RemoteSystems = new ObservableCollection<RemoteSystem>();
+        }
+
+        /// <summary>
+        /// Initiate Picker UI
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task<IEnumerable<RemoteSystem>> PickDeviceAsync()
+        {
+            await ShowAsync();
+            IEnumerable<RemoteSystem> devices = ReturnDevices();
+            return devices;
+        }
+
+        internal IEnumerable<RemoteSystem> ReturnDevices()
+        {
+            List<RemoteSystem> returnValue = new List<RemoteSystem>();
+            foreach (RemoteSystem item in _listDevices.SelectedItems)
+            {
+                returnValue.Add(item);
+            }
+
+            return returnValue.AsEnumerable();
+        }
+
+        /// <inheritdoc />
+        protected override void OnApplyTemplate()
+        {
+            _listDevices = GetTemplateChild("PART_ListDevices") as ListView;
+            _listDeviceTypes = GetTemplateChild("PART_ListDeviceTypes") as ComboBox;
+            _progressRing = GetTemplateChild("PART_ProgressRing") as ProgressRing;
+            _commandSpace = GetTemplateChild("CommandSpace") as Grid;
+
+            List<string> deviceList = typeof(RemoteSystemKinds).GetProperties().Select(a => a.Name).ToList();
+            deviceList.Add("All");
+            deviceList.Add("Unknown");
+
+            _listDeviceTypes.ItemsSource = deviceList.OrderBy(a => a.ToString());
+            _listDeviceTypes.SelectionChanged += ListDeviceTypes_SelectionChanged;
```

Comment:
You need to first unsubscribe old event in case template is changed. For example, the patern is:

```
if (_element != null)
{
// unsubscribe events
}
_element = GetTemplateChild("name of element in template") as TypeOfElement;

if (_element != null)
{
  // set properties, subscribe to events
}